### PR TITLE
Free Trial: Fix Subscription view UI issues

### LINKED ISSUE #8586: [Free Trial] [Test Plan Request] Fix Upgrades screen UI when in landscape mode
Summary
==========
When I am in landscape orientation and I access the Upgrades screen, the button under "troubleshooting" doesn't show its label, even when my phone is using the default font size. I thought that increasing the size might lead to a scrollbar, but it just makes the button itself smaller.

Image

### LINKED ISSUE #8587: [Free Trial] [Test Plan Request] Correctly pluralize Free Trial remaining days message
Summary
==========
When there is 1 day left for a trial it says 1 days instead of 1 day both in the banner and in the upgrades view

Image

### LINKED ISSUE #8589: [Free Trial] [Test Plan Request] Fix upgrades screen text correctness
Summary
==========
On the upgrades screen, there's an extra period at the end of the last sentence ("keep your store running..")

Image

### LINKED ISSUE #8591: [Free Trial] [Test Plan Request] Adjust Upgrades screen to support RTL correctly
Summary
==========
When using RTL language, the message about the free trial in the upgrades screen is sort of mangled - not sure if this will be fixed when actually translated or not.

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Fix issues #8586, #8589, #8587 and add previews for the #8591 scenarios. Since all these issues were minor adjustments, I decided to do them all simultaneously as a single improvement for the Subscription view.

Changelog
==========
- #8586 
Got fixed by adding the vertical scroll Modifier to the View root column so the view doesn't get squeezed with the smaller view height in landscape mode.

- #8587 
Partially fixed through a `pluralizedDays` extension function to the `Period` type. This way, we can generate the correct string translation following localization settings. Tests were added to enforce this behavior too. This improvement will be introduced for the banner in another PR.

- #8589 
Fixed by removing the extra period in the `strings.xml` file. 

- #8591 
Didn't require any fix. The RTL support is working as expected and as mentioned inside the issue comments, the strangeness was probably because of the LTR string. Still, a `@Preview` setting for RTL was added inside the `UpgradesView` so we can adequately verify if any changes we do still keep the expected behavior for RTL mode.

Screenshots
==========
| Before  | After |
| ------------- | ------------- |
| Screenshot_20230321_001338 | Screenshot_20230320_231411 |
| Screenshot_20230321_001346 | Screenshot_20230320_230942 |

How to Test
==========
1. Configure the app with a Free Trial store plan with just one day left in trial.
2. Open the App and go to the bottom bar menu
3. Once there, hit the `Upgrades` button to enter the Subscription view
4. Verify that the `1 days` info now shows as `1 day` as expected.
5. Verify that the additional period in the `...keep your store running.` string is removed
6. Verify that the UI is correctly presented in landscape mode

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/upgrades/UpgradesViewModel.kt [hunk 1]
```diff
@@ -121,6 +76,44 @@ class UpgradesViewModel @Inject constructor(
...
+                name = prettifiedName,
+                freeTrialDuration = FREE_TRIAL_PERIOD,
+                daysLeftInFreeTrial = remainingTrialPeriod.pluralizedDays(resourceProvider)
+            )
+        }
+    }
+
+    private val SitePlan.prettifiedName
+        get() = name.removePrefix("WordPress.com ")
+
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/upgrades/UpgradesViewModel.kt]: A small refactor taking advantage of the added unit tests safety to make the SitePlan evaluation easier to read and calculating the remaining trial period only when it's needed.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/upgrades/UpgradesScreen.kt [hunk 2]
```diff
@@ -143,19 +146,21 @@ fun UpgradesScreen(
 }
 
 @Preview(name = "Light mode")
+@Preview(name = "RTL mode", locale = "ar")
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/upgrades/UpgradesScreen.kt:149]: I couldn't find any problem with the RTL rendering from what was reported in #8591, but I added those previews to check if our changes always support it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/PeriodExt.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import com.woocommerce.android.R
+import com.woocommerce.android.viewmodel.ResourceProvider
+import java.time.Period
+
+fun Period.pluralizedDays(resourceProvider: ResourceProvider): String {
+    return when (days) {
+        1 -> resourceProvider.getString(R.string.free_trial_one_day_left)
+        else -> resourceProvider.getString(R.string.free_trial_days_left_plural, days)
+    }
+}
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/PeriodExt.kt]: WDYT about using `StringUtils#getQuantityString` to keep standard across the app?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/upgrades/UpgradesScreen.kt (hunk 2 above)
Comment (wzieba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/upgrades/UpgradesScreen.kt:149]: > name = "RTL mode"

Wow, nice!

**Review by wzieba — APPROVED** [review]
Looks good! One comment to consider, but nonblocking. Feel free to merge.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/PeriodExt.kt (hunk 3 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/PeriodExt.kt]: Oh, nice one! It's fixed now!

**Pull request merged**